models/discordactions.js

```diff
@@ -505,14 +505,19 @@ const updateUsersNicknameStatus = async (lastNicknameUpdate) => {
         }
       });
 
-      const settledPromises = await Promise.all(promises);
-      nicknameUpdatePromises.push(...settledPromises);
+      const settledPromises = await Promise.allSettled(promises);
+
+      settledPromises.forEach((result) => {
+        if (result.status === "fulfilled" && !!result.value) {
+          successfulUpdates++;
+        } else {
+          logger.error(`Error while updating nickname: ${result.reason}`);
+        }
+      });
```

Comment:
why are we not doing any try, catch-error handling here?
how are we conveying this error to the user? is it somewhere else that I am missing ?

Reply:
> why are we not doing any try, catch-error handling here?
Its present in all the functions

We are okay if 1 out of 100 user nickname update fails. We don't necessarily have to return status 500 for that. We are conveying that using the `successfulUpdates` count in response.

Reply:
But the problem with this approach is that let's say I am on OOO and my update in the status document was done at time 0ms. Say the cron job ran and it tried to update all the available status whose update was done >= 0ms. Now in case my nickname update failed and we allSettle the promise, the nickname update will be successful and the cron job's lastUpdate timestamp will update say to 1ms. Now next time cron job runs, it won't pick up my status. And my OOO will not be marked.

Reply:
- In this case should the cron job handle when it updates the time? 
- In terms of our API, should we send status 500 stating server issues when in reality only one operation failed, out of 100s of mutually exclusive ones?

Reply:
> In this case should the cron job handle when it updates the time?
Can you elaborate as to what you are suggesting?

> In terms of our API, should we send status 500 stating server issues when in reality only one operation failed, out of 100s of mutually exclusive ones?
Ideally, such operations should be a transaction. But maybe we can discuss this more in depth and come to a conclusion.

Reply:
As discussed, we will not be failing the entire call if any of the update for the nickname fails (i.e. this will no longer be a transaction). We will only update the number of calls failed and updated. 
If there were 0 failed updates then in the cron job we will update the `lastNicknameUpdated` timestamp. Else, the timestamp will remain the same.